src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts

```diff
@@ -0,0 +1,50 @@
+import {Directive, EventEmitter, HostListener, Input, Output} from '@angular/core';
+
+export enum LoadDirection {
+    none = 0,
+    top,
+    bottom,
+}
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+})
+export class ScrollWithLoadingDirective {
+    private oldScrollTop = 0;
+    private oldLoadDirection = LoadDirection.none;
+
+    @Input() loadDataBorderOffset = 100;
+
+    @Output() loadData = new EventEmitter<LoadDirection>();
+
+    @HostListener('scroll', ['$event.target'])
+    onScroll(target: HTMLElement) {
+        const {scrollTop, clientHeight, scrollHeight} = target;
+        const scrollBottom = scrollTop + clientHeight;
+        const scrollDif = scrollTop - this.oldScrollTop;
+        const triggerTop = this.loadDataBorderOffset;
+        const triggerBottom = scrollHeight - this.loadDataBorderOffset;
+
+        this.oldScrollTop = scrollTop;
+
+        if (scrollDif) {
+            let newLoadDirection = LoadDirection.none;
+
+            if (scrollDif < 0 && scrollTop <= triggerTop) {
+                newLoadDirection = LoadDirection.top;
+            }
```

Comment:
Рекомендую взять на заметку подход с ранним return, он позволяет строить гибкие ветвления и добавлять промежуточные вычисления, еще и читается отлично - замечательная замена if/else. В данном случае ранний return позволит избежать лишней вложенности кода

```suggestion
        if (!scrollDif) {
            return;
        }
        
        let newLoadDirection = LoadDirection.none;

        if (scrollDif < 0 && scrollTop <= triggerTop) {
            newLoadDirection = LoadDirection.top;
        }
```